doc/tutorial/modes.txt

```diff
@@ -197,6 +197,33 @@ DebugMode      no         yes                VERY HIGH  Make many checks on what
 
 For more detail, see :ref:`Mode<libdoc_compile_mode>` in the library.
 
+.. _optimizers:
+
+Optimizers
+==========
+
+Theano allows compilations with a number of predefined optimizer modes.
```

Comment:
I think the use of the word "mode" here is somewhat confusing. There is compilation modes which are referred to as "modes" in Theano configuration. I would only refer to them as "optimizers", not "optimizer modes". With the description below, I think it's clear enough what the difference between an optimizer and an optimization is.